### PR TITLE
feat: add glue based doubly linked-list

### PR DESCRIPTION
**Pull request by Ashborn-SM**
#### Description of Change
Add glue based doubly linked-list

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C/pull/863"></a>

### CODE REVIEW

**Comment by Ashborn-SM** [comment]
@Panquesito7 please review my PR.

**Comment by Ashborn-SM** [comment]
@Panquesito7 Done all the changes :)

File: data_structures/linked_list/gl_threads.c [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+ * @author [Ashborn-SM](https://github.com/Ashborn-SM)
+ */
+
+#include <stdio.h>  /// for IO operations
+#include <stdlib.h> /// for dynamic memory allocation
+#include <string.h> /// for string manipulation
+#include<assert.h>  /// for Self-testing the implementations
+
+#define HEAD(list) (list->head)
+#define OFFSETOF(structure, field) (long)&((structure*)0)->field
```
Comment (Panquesito7) [inline data_structures/linked_list/gl_threads.c:40]: Please provide documentation on what are these for, how do they work, etc.. 🙂

**Review by Panquesito7 — COMMENTED** [review]
Almost there! 😄

**Pull request closed without merging**